io/io/inc/ROOT/RFile.hxx

```diff
@@ -0,0 +1,204 @@
+/// \file ROOT/RFile.hxx
+/// \ingroup Base ROOT7
+/// \author Giacomo Parolini <giacomo.parolini@cern.ch>
+/// \date 2025-03-19
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+#ifndef ROOT7_RFile
+#define ROOT7_RFile
+
+#include <ROOT/RError.hxx>
+
+#include <memory>
+#include <string_view>
+#include <typeinfo>
+
+class TFile;
+class TKey;
+
+namespace ROOT {
+namespace Experimental {
+
+class RFile;
+struct RFileKeyInfo;
+
+namespace Internal {
+
+ROOT::RLogChannel &RFileLog();
+
+} // namespace Internal
+
+/**
+\class ROOT::Experimental::RFile
+\ingroup RFile
+\brief An interface to read from, or write to, a ROOT file, as well as performing other common operations.
+
+## When and why should you use RFile
+
+RFile is a modern and minimalistic interface to ROOT files, both local and remote, that can be used instead of TFile
+when the following conditions are met:
+- you want a simple interface that makes it easy to do things right and hard to do things wrong;
+- you only need basic Put/Get operations and don't need the more advanced TFile/TDirectory functionalities;
+- you want more robustness and better error reporting for those operations;
+- you want clearer ownership semantics expressed through the type system rather than having objects "automagically"
+  handled for you via implicit ownership of raw pointers.
+
+RFile doesn't try to cover the entirety of use cases covered by TFile/TDirectory/TDirectoryFile and is not
+a 1:1 replacement for them. It is meant to simplify the most common use cases and make them easier to handle by
+minimizing the amount of ROOT-specific quirks and conforming to more standard C++ practices.
```

Comment:
I find the tone a bit harsh.  Consider:
```
RFile is a modern and minimalistic interface to ROOT files, both local and remote, that can be used instead of TFile
when you only need basic Put/Get operations and don't need the more advanced TFile/TDirectory functionalities.  It provides:
- a simple interface that makes it easy to do things right and hard to do things wrong,
- more robustness and better error reporting for those operations,
- clearer ownership semantics expressed through the type system.

RFile doesn't cover the entirety of use cases covered by TFile/TDirectory/TDirectoryFile and is not
a 1:1 replacement for them.  It is meant to simplify the most common use cases by following newer standard C++ practices.
```